snirf_specification.md

```diff
@@ -560,6 +560,119 @@ As described below, optional variables `probe.sourceLabels` and
 `probe.detectorLabels` are provided for indicating the instrument specific 
 label for sources and detectors.
 
+#### /nirs(i)/data(j)/measurementLists
+* **Presence**: optional
+* **Type**:  group
+* **Location**: `/nirs(i)/data(j)/measurementLists`
+
+The group for measurement list variables which map data array onto the to map the data array onto the probe geometry (sources and detectors), data type, and wavelength. This group's datasets are arrays with size `<number of channels>`, which each position describing the corresponding column in the data matrix. (i.e. the values at `measurementLists/sourceIndex(3)` and `measurementLists/detectorIndex(3)` correspond to `dataTimeSeries(:,3)`).
+
+This group is required only if the indexed-group format `/nirs(i)/data(j)/measurementList(k)` is not used to encode the measurement list. `measurementLists` is an alternative that may offer better performance for larger probes.
+
+The arrays of `measurementLists` are:
+
+#### /nirs(i)/data(j)/measurementLists/sourceIndex 
+* **Presence**: required
```

Comment:
for all the places in this chunk of code where we say "required", do we need to clarify that this is "required if measurementLists is utilized?

Reply:
It seems like this could be merged... but @sstucker is not working on this anymore. Can someone else make the fixes in the review comments?

Also, we need a to do list item to remind us that the validator needs to incorporate this. Do we just create a new Issue for getting that done? But the validator is not in this GitHub... hmm... I forget who is managing the validator.

Reply:
I can update the validator logic in pysnirf2 as I recall working with files like this and have some other pending work there I want to release.  I have the bandwidth for this, no worries

I do not feel comfortable finishing this feature for the specification itself.

There are some issues with this feature as I drafted it here, for instance, how do we deal with the missing values in the table? Should we delineate a column order for all possible channel datasets? Are NAN values supported by HDF5 and portable to all commonly used interfaces?

Reply:
@sstucker I am fine with measurementLists and am not confused by this being one letter different than measurementList.
I do wonder why we just didn't utilize the existing measurementList(1) for this as I described way back in https://github.com/fNIRS/snirf/issues/103#issuecomment-1073038956, but I can live with measurementLists as I appreciate that it makes the intent of using arrays for describing the measurementList explicit.